WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/Visibility/ProductVisibilityViewController.swift

```diff
@@ -0,0 +1,323 @@
+            comment: "Alert message to confirm the user wants to discard changes in Product Visibility"
+        )
+
+        let alertController = UIAlertController(title: messageTitle, message: messageDescription, preferredStyle: .alert)
+
+        let cancelText = NSLocalizedString("Discard", comment: "Alert button title - dismisses alert, which discard changes on Product Visibility screen")
+        alertController.addActionWithTitle(cancelText, style: .cancel) { [weak self] _ in
+            self?.navigationController?.popViewController(animated: true)
+        }
+
+        let logoutText = NSLocalizedString("Keep Editing", comment: "Alert button title - which keeps the user on the Product Visibility screen")
+        alertController.addDefaultActionWithTitle(logoutText) { _ in
+        }
+
+        present(alertController, animated: true)
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension ProductVisibilityViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = sections[indexPath.section].rows[indexPath.row]
+
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension ProductVisibilityViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        let row = sections[indexPath.section].rows[indexPath.row]
+        visibility = row.visibility
+        productSettings.status = getProductStatus(visibility)
+        reloadSections()
+    }
+}
+
+// MARK: - Support for UITableViewDataSource
+//
+private extension ProductVisibilityViewController {
+
+    /// Configure cellForRowAtIndexPath:
+    ///
+   func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as BasicTableViewCell:
+            configureVisibilityCell(cell: cell, indexPath: indexPath)
+        case let cell as TitleAndTextFieldWithImageTableViewCell:
+            configurePasswordFieldCell(cell: cell, indexPath: indexPath)
+        default:
+            fatalError("Unidentified product visibility row type")
+        }
+    }
+
+    func configureVisibilityCell(cell: BasicTableViewCell, indexPath: IndexPath) {
+        let row = sections[indexPath.section].rows[indexPath.row]
+        cell.selectionStyle = .default
+        cell.textLabel?.text = row.description
+        if row.editable {
+            cell.accessoryType = .none
```

Comment:
There is a cell that is editable, the `passwordField` row 👍

Reply:
hmm from my understanding, the `passwordField` row would go to call `configurePasswordFieldCell` instead of this function so that this function configures the visibility cells with a checkmark?